Makefile

```diff
@@ -0,0 +1,53 @@
+# Makefile for running typical developer workflow actions.
+# To run actions in a subdirectory of the repo:
+#   make lint build dir=translate/snippets
+#
+# Python version may also be specified in the command as 'py=3.10'
+
+
+# Default values for make variables:
+# dir will use repo root as working directory if not specified.
+dir ?= $(shell pwd)
+# python version: defaults to 3.11
+py ?= "3.11"
+
+INTERFACE_ACTIONS="build test lint"
+repo_root = $(shell pwd)
+.ONESHELL: #ease subdirectory work by using the same subshell for all commands
+.-PHONY: *
+
+# Export env vars used to determine cloud project.
+export GOOGLE_CLOUD_PROJECT ?= ${GOOGLE_SAMPLE_PROJECT}
+export BUILD_SPECIFIC_GCLOUD_PROJECT ?= ${GOOGLE_SAMPLE_PROJECT}
+
+build:
+	pip install nox
+	cd ${dir}
+	pip install -r requirements.txt
+
+test: check-env build noxfile.py
+	cd ${dir}
+	nox $(NOXFLAGS) -s py-$(py)
+
+lint: noxfile.py
+	pip install nox black
+	cd ${dir}
+	nox $(NOXFLAGS) -s blacken
+	nox $(NOXFLAGS) -s lint
+
+# if no noxfile is present, we create one from the toplevel noxfile-template.py
+noxfile.py:
+	cd ${dir}
+	cp ${repo_root}/noxfile-template.py noxfile.py
```

Comment:
I think this will just overwrite the existing noxfile.py, no matter if it's there or not. I think you want to add `-n` flag here.

Reply:
I *think* Make will do the right thing when a noxfile.py exists already, but there's no harm in the -n, so i've added it anyway just to be safe